--- sequelize_old.sql	2023-07-05 21:40:06.926984271 +0200
+++ sequelize_typescript.sql	2023-07-10 13:40:45.597516784 +0200
@@ -198,9 +198,9 @@
     "pageId" uuid,
     "communityId" uuid,
     "scopeSummaryId" uuid,
+    "crossrefDepositRecordId" uuid,
     "createdAt" timestamp with time zone NOT NULL,
-    "updatedAt" timestamp with time zone NOT NULL,
-    "crossrefDepositRecordId" uuid
+    "updatedAt" timestamp with time zone NOT NULL
 );
 
 
@@ -273,9 +273,9 @@
     "defaultPubCollections" jsonb,
     "spamTagId" uuid,
     "organizationId" uuid,
+    "scopeSummaryId" uuid,
     "createdAt" timestamp with time zone NOT NULL,
-    "updatedAt" timestamp with time zone NOT NULL,
-    "scopeSummaryId" uuid
+    "updatedAt" timestamp with time zone NOT NULL
 );
 
 
@@ -815,10 +815,10 @@
     "commentHash" character varying(255),
     "draftId" uuid NOT NULL,
     "communityId" uuid NOT NULL,
-    "createdAt" timestamp with time zone NOT NULL,
-    "updatedAt" timestamp with time zone NOT NULL,
     "crossrefDepositRecordId" uuid,
-    "scopeSummaryId" uuid
+    "scopeSummaryId" uuid,
+    "createdAt" timestamp with time zone NOT NULL,
+    "updatedAt" timestamp with time zone NOT NULL
 );
 
 
@@ -1198,8 +1198,7 @@
     "userId" uuid NOT NULL,
     "visibilityId" uuid NOT NULL,
     "createdAt" timestamp with time zone NOT NULL,
-    "updatedAt" timestamp with time zone NOT NULL,
-    "UserId" uuid
+    "updatedAt" timestamp with time zone NOT NULL
 );
 
 
@@ -1233,10 +1232,10 @@
     id uuid NOT NULL,
     "zoteroUsername" text,
     "zoteroUserId" text,
-    "createdAt" timestamp with time zone NOT NULL,
-    "updatedAt" timestamp with time zone NOT NULL,
     "userId" uuid NOT NULL,
-    "integrationDataOAuth1Id" uuid NOT NULL
+    "integrationDataOAuth1Id" uuid NOT NULL,
+    "createdAt" timestamp with time zone NOT NULL,
+    "updatedAt" timestamp with time zone NOT NULL
 );
 
 
@@ -1278,7 +1277,7 @@
 -- Data for Name: Collections; Type: TABLE DATA; Schema: public; Owner: testuser
 --
 
-COPY public."Collections" (id, title, slug, avatar, "isRestricted", "isPublic", "viewHash", "editHash", metadata, kind, doi, "readNextPreviewSize", layout, "layoutAllowsDuplicatePubs", "pageId", "communityId", "scopeSummaryId", "createdAt", "updatedAt", "crossrefDepositRecordId") FROM stdin;
+COPY public."Collections" (id, title, slug, avatar, "isRestricted", "isPublic", "viewHash", "editHash", metadata, kind, doi, "readNextPreviewSize", layout, "layoutAllowsDuplicatePubs", "pageId", "communityId", "scopeSummaryId", "crossrefDepositRecordId", "createdAt", "updatedAt") FROM stdin;
 \.
 
 
@@ -1294,7 +1293,7 @@
 -- Data for Name: Communities; Type: TABLE DATA; Schema: public; Owner: testuser
 --
 
-COPY public."Communities" (id, subdomain, domain, title, "citeAs", "publishAs", description, avatar, favicon, "accentColorLight", "accentColorDark", "hideCreatePubButton", "headerLogo", "headerLinks", "headerColorType", "useHeaderTextAccent", "hideHero", "hideHeaderLogo", "heroLogo", "heroBackgroundImage", "heroBackgroundColor", "heroTextColor", "useHeaderGradient", "heroImage", "heroTitle", "heroText", "heroPrimaryButton", "heroSecondaryButton", "heroAlign", navigation, "hideNav", "navLinks", "footerLinks", "footerLogoLink", "footerTitle", "footerImage", website, facebook, twitter, email, issn, "isFeatured", "viewHash", "editHash", "premiumLicenseFlag", "defaultPubCollections", "spamTagId", "organizationId", "createdAt", "updatedAt", "scopeSummaryId") FROM stdin;
+COPY public."Communities" (id, subdomain, domain, title, "citeAs", "publishAs", description, avatar, favicon, "accentColorLight", "accentColorDark", "hideCreatePubButton", "headerLogo", "headerLinks", "headerColorType", "useHeaderTextAccent", "hideHero", "hideHeaderLogo", "heroLogo", "heroBackgroundImage", "heroBackgroundColor", "heroTextColor", "useHeaderGradient", "heroImage", "heroTitle", "heroText", "heroPrimaryButton", "heroSecondaryButton", "heroAlign", navigation, "hideNav", "navLinks", "footerLinks", "footerLogoLink", "footerTitle", "footerImage", website, facebook, twitter, email, issn, "isFeatured", "viewHash", "editHash", "premiumLicenseFlag", "defaultPubCollections", "spamTagId", "organizationId", "scopeSummaryId", "createdAt", "updatedAt") FROM stdin;
 \.
 
 
@@ -1407,7 +1406,7 @@
 --
 
 COPY public."FeatureFlags" (id, name, "enabledUsersFraction", "enabledCommunitiesFraction", "createdAt", "updatedAt") FROM stdin;
-959c82da-39c3-460b-8a69-7ac3a80d1db1	customScripts	0	0	2023-07-05 21:39:50.861+02	2023-07-05 21:39:50.861+02
+b1318338-392c-4765-b108-7a200f512b40	customScripts	0	0	2023-07-10 13:40:42.99+02	2023-07-10 13:40:42.99+02
 \.
 
 
@@ -1535,7 +1534,7 @@
 -- Data for Name: Pubs; Type: TABLE DATA; Schema: public; Owner: testuser
 --
 
-COPY public."Pubs" (id, slug, title, "htmlTitle", description, "htmlDescription", avatar, "customPublishedAt", doi, labels, downloads, metadata, "viewHash", "editHash", "reviewHash", "commentHash", "draftId", "communityId", "createdAt", "updatedAt", "crossrefDepositRecordId", "scopeSummaryId") FROM stdin;
+COPY public."Pubs" (id, slug, title, "htmlTitle", description, "htmlDescription", avatar, "customPublishedAt", doi, labels, downloads, metadata, "viewHash", "editHash", "reviewHash", "commentHash", "draftId", "communityId", "crossrefDepositRecordId", "scopeSummaryId", "createdAt", "updatedAt") FROM stdin;
 \.
 
 
@@ -1695,7 +1694,7 @@
 -- Data for Name: VisibilityUsers; Type: TABLE DATA; Schema: public; Owner: testuser
 --
 
-COPY public."VisibilityUsers" (id, "userId", "visibilityId", "createdAt", "updatedAt", "UserId") FROM stdin;
+COPY public."VisibilityUsers" (id, "userId", "visibilityId", "createdAt", "updatedAt") FROM stdin;
 \.
 
 
@@ -1711,7 +1710,7 @@
 -- Data for Name: ZoteroIntegrations; Type: TABLE DATA; Schema: public; Owner: testuser
 --
 
-COPY public."ZoteroIntegrations" (id, "zoteroUsername", "zoteroUserId", "createdAt", "updatedAt", "userId", "integrationDataOAuth1Id") FROM stdin;
+COPY public."ZoteroIntegrations" (id, "zoteroUsername", "zoteroUserId", "userId", "integrationDataOAuth1Id", "createdAt", "updatedAt") FROM stdin;
 \.
 
 
@@ -2236,11 +2235,11 @@
 
 
 --
--- Name: VisibilityUsers VisibilityUsers_visibilityId_UserId_key; Type: CONSTRAINT; Schema: public; Owner: testuser
+-- Name: VisibilityUsers VisibilityUsers_userId_visibilityId_key; Type: CONSTRAINT; Schema: public; Owner: testuser
 --
 
 ALTER TABLE ONLY public."VisibilityUsers"
-    ADD CONSTRAINT "VisibilityUsers_visibilityId_UserId_key" UNIQUE ("visibilityId", "UserId");
+    ADD CONSTRAINT "VisibilityUsers_userId_visibilityId_key" UNIQUE ("userId", "visibilityId");
 
 
 --
@@ -2260,6 +2259,27 @@
 
 
 --
+-- Name: CollectionPub_collectionId_pubId; Type: INDEX; Schema: public; Owner: testuser
+--
+
+CREATE UNIQUE INDEX "CollectionPub_collectionId_pubId" ON public."CollectionPubs" USING btree ("pubId", "collectionId");
+
+
+--
+-- Name: UserScopeVisit_userId_collectionId; Type: INDEX; Schema: public; Owner: testuser
+--
+
+CREATE UNIQUE INDEX "UserScopeVisit_userId_collectionId" ON public."UserScopeVisits" USING btree ("userId", "collectionId");
+
+
+--
+-- Name: UserScopeVisit_userId_pubId; Type: INDEX; Schema: public; Owner: testuser
+--
+
+CREATE UNIQUE INDEX "UserScopeVisit_userId_pubId" ON public."UserScopeVisits" USING btree ("userId", "pubId");
+
+
+--
 -- Name: activity_items_actor_id; Type: INDEX; Schema: public; Owner: testuser
 --
 
@@ -2288,13 +2308,6 @@
 
 
 --
--- Name: collection_pubs_collection_id_pub_id; Type: INDEX; Schema: public; Owner: testuser
---
-
-CREATE UNIQUE INDEX collection_pubs_collection_id_pub_id ON public."CollectionPubs" USING btree ("collectionId", "pubId");
-
-
---
 -- Name: discussion_anchors_discussion_id; Type: INDEX; Schema: public; Owner: testuser
 --
 
@@ -2400,20 +2413,6 @@
 
 
 --
--- Name: user_scope_visits_user_id_collection_id; Type: INDEX; Schema: public; Owner: testuser
---
-
-CREATE UNIQUE INDEX user_scope_visits_user_id_collection_id ON public."UserScopeVisits" USING btree ("userId", "collectionId");
-
-
---
--- Name: user_scope_visits_user_id_pub_id; Type: INDEX; Schema: public; Owner: testuser
---
-
-CREATE UNIQUE INDEX user_scope_visits_user_id_pub_id ON public."UserScopeVisits" USING btree ("userId", "pubId");
-
-
---
 -- Name: user_subscriptions_pub_id; Type: INDEX; Schema: public; Owner: testuser
 --
 
@@ -2487,7 +2486,7 @@
 --
 
 ALTER TABLE ONLY public."Collections"
-    ADD CONSTRAINT "Collections_crossrefDepositRecordId_fkey" FOREIGN KEY ("crossrefDepositRecordId") REFERENCES public."CrossrefDepositRecords"(id) ON UPDATE CASCADE ON DELETE SET NULL;
+    ADD CONSTRAINT "Collections_crossrefDepositRecordId_fkey" FOREIGN KEY ("crossrefDepositRecordId") REFERENCES public."CrossrefDepositRecords"(id) ON UPDATE CASCADE;
 
 
 --
@@ -2503,7 +2502,7 @@
 --
 
 ALTER TABLE ONLY public."Collections"
-    ADD CONSTRAINT "Collections_scopeSummaryId_fkey" FOREIGN KEY ("scopeSummaryId") REFERENCES public."ScopeSummaries"(id) ON UPDATE CASCADE ON DELETE SET NULL;
+    ADD CONSTRAINT "Collections_scopeSummaryId_fkey" FOREIGN KEY ("scopeSummaryId") REFERENCES public."ScopeSummaries"(id) ON UPDATE CASCADE;
 
 
 --
@@ -2519,7 +2518,7 @@
 --
 
 ALTER TABLE ONLY public."Communities"
-    ADD CONSTRAINT "Communities_scopeSummaryId_fkey" FOREIGN KEY ("scopeSummaryId") REFERENCES public."ScopeSummaries"(id) ON UPDATE CASCADE ON DELETE SET NULL;
+    ADD CONSTRAINT "Communities_scopeSummaryId_fkey" FOREIGN KEY ("scopeSummaryId") REFERENCES public."ScopeSummaries"(id) ON UPDATE CASCADE;
 
 
 --
@@ -2831,7 +2830,7 @@
 --
 
 ALTER TABLE ONLY public."Pubs"
-    ADD CONSTRAINT "Pubs_crossrefDepositRecordId_fkey" FOREIGN KEY ("crossrefDepositRecordId") REFERENCES public."CrossrefDepositRecords"(id) ON UPDATE CASCADE ON DELETE SET NULL;
+    ADD CONSTRAINT "Pubs_crossrefDepositRecordId_fkey" FOREIGN KEY ("crossrefDepositRecordId") REFERENCES public."CrossrefDepositRecords"(id) ON UPDATE CASCADE;
 
 
 --
@@ -2839,7 +2838,7 @@
 --
 
 ALTER TABLE ONLY public."Pubs"
-    ADD CONSTRAINT "Pubs_draftId_fkey" FOREIGN KEY ("draftId") REFERENCES public."Drafts"(id) ON UPDATE CASCADE;
+    ADD CONSTRAINT "Pubs_draftId_fkey" FOREIGN KEY ("draftId") REFERENCES public."Drafts"(id) ON UPDATE CASCADE ON DELETE CASCADE;
 
 
 --
@@ -2847,7 +2846,7 @@
 --
 
 ALTER TABLE ONLY public."Pubs"
-    ADD CONSTRAINT "Pubs_scopeSummaryId_fkey" FOREIGN KEY ("scopeSummaryId") REFERENCES public."ScopeSummaries"(id) ON UPDATE CASCADE ON DELETE SET NULL;
+    ADD CONSTRAINT "Pubs_scopeSummaryId_fkey" FOREIGN KEY ("scopeSummaryId") REFERENCES public."ScopeSummaries"(id) ON UPDATE CASCADE;
 
 
 --
@@ -3035,11 +3034,11 @@
 
 
 --
--- Name: VisibilityUsers VisibilityUsers_UserId_fkey; Type: FK CONSTRAINT; Schema: public; Owner: testuser
+-- Name: VisibilityUsers VisibilityUsers_userId_fkey; Type: FK CONSTRAINT; Schema: public; Owner: testuser
 --
 
 ALTER TABLE ONLY public."VisibilityUsers"
-    ADD CONSTRAINT "VisibilityUsers_UserId_fkey" FOREIGN KEY ("UserId") REFERENCES public."Users"(id) ON UPDATE CASCADE ON DELETE CASCADE;
+    ADD CONSTRAINT "VisibilityUsers_userId_fkey" FOREIGN KEY ("userId") REFERENCES public."Users"(id) ON UPDATE CASCADE ON DELETE CASCADE;
 
 
 --
